edge-hub/src/Microsoft.Azure.Devices.Routing.Core/RouteResult.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Routing.Core
+{
+    using System;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class RouteResult : IEquatable<RouteResult>
+    {
+        public RouteResult(Endpoint endpoint, uint priority, uint timeToLiveSecs)
+        {
+            this.Endpoint = Preconditions.CheckNotNull(endpoint, nameof(endpoint));
+            this.Priority = priority;
+            this.TimeToLiveSecs = timeToLiveSecs;
+        }
+
+        public Endpoint Endpoint { get; }
+        public uint Priority { get; }
+        public uint TimeToLiveSecs { get; }
+
+        public bool Equals(RouteResult other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            // Check endpoint, priority, and TTL
+            bool areEqual =
+                this.Endpoint.Equals(other.Endpoint) &&
+                this.Priority == other.Priority &&
+                this.TimeToLiveSecs == other.TimeToLiveSecs;
+
+            return areEqual;
+        }
+
+        public override bool Equals(object obj) => obj.GetType() == this.GetType() && this.Equals((RouteResult)obj);
+
+        public override int GetHashCode()
+        {
+            // Not accurate to cast from uint to int, but for hashing purposes
+            // that's okay, and much better than getting an int overflow
+            unchecked
+            {
+                return this.Endpoint.Id.GetHashCode() + (int)this.Priority + (int)this.TimeToLiveSecs;

```

Comment:
Should we find a better way to increase the uniqueness of hash code value?  Normally a prime number will involve in this calculation.

Reply:
I actually think this is desirable, for a given endpoint, if the priority and TTL are the same, then the two results are effectively the same thing.  The hash from the underlying endpoint provides enough uniqueness for all other comparisons.

---
In reply to: [377450086](https://github.com/Azure/iotedge/pull/2469#discussion_r377450086) [](ancestors = 377450086)